docs/developer/designs/topology-awareness/README.md

```diff
@@ -0,0 +1,287 @@
+# Topology Aware Scheduling
+
+## Summary
+
+This design document outlines an approach for implementing topology aware scheduling in KAI-Scheduler, leveraging the topology CRD concepts from Kueue. Topology awareness enables more efficient scheduling of workloads by considering the physical placement of resources within a cluster, leading to improved performance for applications that benefit from resource locality.
+
+## Motivation
+
+Workloads with specific topology requirements, such as distributed training jobs or applications sensitive to inter-node communication latency, benefit significantly from intelligent placement decisions that respect the underlying hardware topology. Without topology awareness, jobs may be scheduled across physically distant nodes, leading to:
+
+- Increased network latency and reduced bandwidth
+- Suboptimal performance for compute-intensive workloads
+- Inefficient resource utilization across the cluster
+
+## Goals
+
+- Implement topology-aware scheduling in KAI-Scheduler
+- Leverage Kueue's topology CRD concepts to keep an industry standard
+   - Support hierarchical topology representations (e.g., datacenter/rack/node/device)
+- Provide flexible configuration options for topology-based placement
+- Support both Required and Preferred topology constraint request
+
+## Non-Goals
+
+- Automatically detecting topology without user input
+- Ensure backward compatibility with existing scheduling mechanisms
+- Fixing KAI-scheduler scheduling issues with Jobs that use PodAffinity assignments
```

Comment:
Can you explain the second point?

Also, I am not sure that results based on PodAffinity is an issue, it simply can lead to a sub-optimal result and topology aware scheduling is the way to achieve a better result. I can see how in most cases these features have different goals. What do you think?